include/envoy/server/options.h

```diff
@@ -127,6 +128,11 @@ class Options {
    */
   virtual const std::string& serviceZone() PURE;
 
+  /**
+   * @return const std::vector<Stats::Tag> default stats tags.
+   */
+  virtual const std::vector<Stats::Tag> statsTags() PURE;
```

Comment:
I'm thinking about whether it is better to return and handle `Tag`s via pointer (with unique_ptr?) here and rest of this PR.

Reply:
Why can't this be a `const std::vector<Stats::Tag>&` return? I don't think a single copy when this later gets used in the tag bucket is particularly bad at startup.

Reply:
Right. I'm working for that.